amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -184,29 +198,55 @@ private void handleFlowError(Throwable t) {
         }
     }
 
-    private String throwableCategory(Throwable t) {
+    private enum ThrowableType {
+        ACUIRE_TIMEOUT("AcquireTimeout"), READ_TIMEOUT("ReadTimeout"), UNKNOWN("Unknown");
+
+        String value;
+
+        ThrowableType(final String value) {
+            this.value = value;
+        }
+    }
+
+    private class ThrowableCategory {
+        @NonNull
+        final ThrowableType throwableType;
+        @NonNull
+        final String throwableTypeString;
+
+        ThrowableCategory(final ThrowableType throwableType) {
+            this(throwableType, throwableType.value);
+        }
+
+        ThrowableCategory(final ThrowableType throwableType, final String throwableTypeString) {
+            this.throwableType = throwableType;
+            this.throwableTypeString = throwableTypeString;
+        }
+    }
+
+    private ThrowableCategory throwableCategory(Throwable t) {
         Throwable current = t;
         StringBuilder builder = new StringBuilder();
         do {
             if (current.getMessage() != null && current.getMessage().startsWith("Acquire operation")) {
-                return "AcquireTimeout";
+                return new ThrowableCategory(ThrowableType.ACUIRE_TIMEOUT);
```

Comment:
These should just be static instances of the ThrowableCategory as they have no local state.

Reply:
Addressed